pkg/basic_server/basic_server.go

```diff
@@ -27,14 +28,18 @@ type Server interface {
 	Name() string
 	// Context returns the context of server.
 	Context() context.Context
-
 	// Run runs the server.
 	Run() error
 	// Close closes the server.
 	Close()
-
 	// GetClient returns builtin etcd client.
 	GetClient() *clientv3.Client
 	// GetHTTPClient returns builtin etcd client.
 	GetHTTPClient() *http.Client
+	// AddStartCallback adds a callback in the startServer phase.
+	AddStartCallback(callbacks ...func())
+	// GetMember returns the member information.
+	GetMember() *member.Member
+	// AddLeaderCallback adds a callback in the leader campaign phase.
+	AddLeaderCallback(callbacks ...func(context.Context))
```

Comment:
Do we need to put this API into basicserver.Server? This API means that this server will elect for leader, correct? It doesn't sound a must-havior behavior of a **basic** server.

Reply:
I also consider to replace it with primary

Reply:
add todo